### ISSUE TITLE
Save and Close button on Next State doesn't work

### ISSUE THREAD

**Issue by i-m-foster** [amgrubb/BloomingLeaf#660]
The Save and Close Button seems to just regenerate a new path without taking the saved satisfaction values from Next State into account

**Comment by amgrubb** [comment]
I believe the error is in the `initializePrevResults` method in CSPNode.java:
```
	public static void initializePrevResults(ModelSpec spec, List<Constraint> constraints, 
			IntVar[] timePoints, BooleanVar[][][] values,
			HashMap<String, Integer> uniqueIDToValueIndex) {
```
The previous values are not assigned at the end of the function.

**Comment by amgrubb** [comment]
I believe I've fixed this bug, I'm going to try and also fix #575 before creating a pull request.

**Comment by amgrubb** [comment]
_Note:_ my updated sweet alert in `next-state.js` (line ~1148) is not appearing because the window closes first. @YesugenB & @i-m-foster do you know how to pause the call to `save_current_state()` until the user clicks ok?
